gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/ApiReactorHandlerFactory.java

```diff
@@ -107,19 +108,23 @@ public class ApiReactorHandlerFactory implements ReactorHandlerFactory<Api> {
     private final ApiProcessorChainFactory apiProcessorChainFactory;
     private final OrganizationManager organizationManager;
     private final FlowResolverFactory flowResolverFactory;
+    private final Long httpRequestTimeout;
+    private final Long httpRequestTimeoutGraceDelay;
     private ApplicationContext applicationContext;
 
     public ApiReactorHandlerFactory(
         ApplicationContext applicationContext,
         Configuration configuration,
         Node node,
-        io.gravitee.gateway.policy.PolicyFactoryCreator v3PolicyFactoryCreator,
+        PolicyFactoryCreator v3PolicyFactoryCreator,
         io.gravitee.gateway.jupiter.policy.PolicyFactory policyFactory,
         io.gravitee.gateway.jupiter.policy.PolicyChainFactory platformPolicyChainFactory,
         OrganizationManager organizationManager,
         PolicyChainProviderLoader policyChainProviderLoader,
         ApiProcessorChainFactory apiProcessorChainFactory,
-        FlowResolverFactory flowResolverFactory
+        FlowResolverFactory flowResolverFactory,
+        Long httpRequestTimeout,
+        Long httpRequestTimeoutGraceDelay
```

Comment:
Is it possible to get them from the configuration like it has been done for other tracing or pending request timeout  ? This will avoid having a constructor that keeps going bigger 🙂

Reply:
resolved by [23d903c](https://github.com/gravitee-io/gravitee-api-management/pull/2224/commits/23d903cbace24f58a27b5caac671cfd359bf9d83)